docs/developer-guide/udf.rst

```diff
@@ -682,12 +688,188 @@ You can use this to better describe what a particular version of the UDF does, f
     public static Udaf<String, Struct, Double> averageStringLength(final String initialString){...}
 
 
+.. _ksql-udtfs:
+
+UDTFs
+-----
+
+To create a UDTF you need to create a class that is annotated with ``@UdtfDescription``.
+Each method in the class that represents a UDTF must be public and annotated with ``@Udtf``. The class
+you create represents a collection of UDTFs all with the same name but may have different
+arguments and return types.
+
+``@UdfParameter`` annotations can be added to method parameters to provide users with richer
+information, including the parameter schema. This annotation is required if the KSQL type cannot
+be inferred from the Java type (e.g. ``STRUCT``).
+
+
+Null Handling
+~~~~~~~~~~~~~
+
+If a UDTF uses primitive types in its signature it is indicating that the parameter should never be null.
```

Comment:
```suggestion
If a UDTF uses primitive types in its signature it is indicating that the parameter should never be null.
```
```suggestion
If a UDTF uses primitive types in its signature, this indicates that the parameter should never be null.
```

Reply:
this is just copy and paste from the UDF section :)